.github/workflows/build-and-test.yml

```
@@ -0,0 +1,116 @@
+                    - framework: netcoreapp3.0
+                      sdk_version: 3.0.x
+                    - framework: netcoreapp3.1
+                      sdk_version: 3.1.x
+                    - framework: net5.0
+                      sdk_version: 5.0.x
+                    - framework: net6.0
+                      sdk_version: 6.0.x
+                    - framework: net7.0
+                      sdk_version: 7.0.x
+                    - framework: net8.0
+                      sdk_version: 8.0.x
+                    - framework: net9.0
+                      sdk_version: 9.0.x
+                    - framework: net10.0
+                      sdk_version: 10.0.x
+                    - framework: net48
+                      sdk_version: 10.0.x
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+
+            - name: Install libssl on Linux
+              # .NET 5.0 and below want libssl v1, which is not available on Ubuntu 24.04 normally
+              if: runner.os == 'Linux'
+              run: |
+                  curl -O https://security.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.24_amd64.deb
+                  sudo dpkg -i libssl1.1_1.1.1f-1ubuntu2.24_amd64.deb
+
+            - name: Setup .NET SDK 10.0.x
+              # Always install SDK 10.0.x for building
+              uses: actions/setup-dotnet@v5
+              with:
+                  dotnet-version: 10.0.x
+
+            - name: Setup .NET SDK ${{ matrix.sdk_version }}
+              # Install the version-specific .NET SDK because need the matching runtime version to run tests
+              # (ideally we would install only the runtime, not the full SDK, but `actions/setup-dotnet` can only install SDKs)
+              if: matrix.sdk_version != '10.0.x'
+              uses: actions/setup-dotnet@v5
+              with:
+                  dotnet-version: ${{ matrix.sdk_version }}
+
+            - name: Run tests
+              run: dotnet test --blame-hang --blame-hang-dump-type none --blame-hang-timeout 60s --framework ${{ matrix.framework }}
+
+    build-nuget:
+        name: Build NuGet Package
+        runs-on: ubuntu-latest
+        permissions:
+            actions: read
+            contents: read
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+
+            - name: Setup .NET SDK 10.0.x
+              uses: actions/setup-dotnet@v5
+              with:
+                  dotnet-version: 10.0.x
+
+            - name: Build NuGet package
+              run: dotnet pack src/StatsdClient/StatsdClient.csproj --configuration Release --output ./nuget-packages
+
+            - name: Upload NuGet package artifact
+              uses: actions/upload-artifact@v4
+              with:
+                  name: nuget-package
+                  path: ./nuget-packages/*.*nupkg
+                  if-no-files-found: error
+                  retention-days: 30
```

Maybe controversial, but is there any point building this _again_, in another runner? We could also just only build it in the .NET 10 branch (for example?). Optional, but reduces usage 🤷‍♂️ Either way, make sure to pin dependencies 🙂 

```suggestion
            - name: Build NuGet package
              if: matrix.framework == 'net10.0' && matrix.os == 'ubuntu-latest'
              run: dotnet pack src/StatsdClient/StatsdClient.csproj --configuration Release --output ./nuget-packages

            - name: Upload NuGet package artifact
              if: matrix.framework == 'net10.0' && matrix.os == 'ubuntu-latest'
              uses: actions/upload-artifact@v4
              with:
                  name: nuget-package
                  path: ./nuget-packages/*.*nupkg
                  if-no-files-found: error
                  retention-days: 30
```

---

Also, if you use the artifacts output layout (see later) you can omit the `--output ./nuget-packages` and just pull from the default location `./artifacts/packages/*.nupkg`

---

Nice, thanks!

---

```
uses: actions/upload-artifact@v4
```
👀